docs/rfcs/2025-01-20-member-mentions.md

```diff
@@ -4,36 +4,57 @@
 
 Mention members in the messages.
 
+There are several UX objectives that mentions must deliver:
+- to notify the user about the messages that mention the user or to reply to user's messages - groups must have this notification mode (already present in the API),
+- to allow the user to navigate to unread mentions and replies,
+- to highlight mentions and allow all users to open mentioned member profile - this is the least important objective.
+
 ## Solution
 
-Message text can include references to the shared member IDs in base64 format:
+Message text should include the reference to the shared member display name:
+
+```
+Hello @name
+```
+
+or
 
 ```
-Hello @'abcd1234'
+Hello @'member name'
 ```
 
-In addition, the message JSON will include the array of mentions, as objects with property `memberId`. This is to ensure the intent and that the fragments of text are treated as mentions.
+This is the same format that people already use and that is currently supported in the API. The name in the message should use the display name at the time of mention, for backwards compatibility, and the message should additionally the mapping from used display names to shared group member IDs, and the UI would show the current display name (at the time of loading the message).
+
+For this mapping the message JSON will include the array of mentions, as objects with properties `displayName` and `memberId`. This is to ensure the intent and that the fragments of text are treated as mentions.
 
 Using an immutable `memberId` would prevent any race conditions and duplicate display names. The receiving client would show a local view name (display name or an alias), and would open a correct member card when mention is tapped.
 
-There should be a reasonable limit on the number of mentions in a group, e.g. 3. This is to prevent abuse, expensive processing both in the client and in super-peers that would have to forward member profiles if they were not forwarded before. This limit has to be enforced both on sending and receiving ends.
+As display names are not unique in the group, we should convert them to locally-unique names (per message), by appending _1, _2, as necessary, and the same locally unique names should be used in the mapping to member IDs. These locally unique names must NOT use local user aliases, and must NOT use localDisplayName, as otherwise it may leak information that is known only to the user's client.
+
+There should be a reasonable limit on the number of mentions per message, e.g. 3. This is to prevent abuse, expensive processing both in the client and in super-peers that would have to forward member profiles if they were not forwarded before. This limit has to be enforced both on sending and receiving ends.
```

Comment:
it's not much harder to send multiple messages with different mentions, so what's the point in this limitation? also why does profile need to be forwarded if mention now includes name?

Reply:
not profile, mapping from mentioned name to member_id